src/App.tsx

```diff
@@ -1,11 +1,70 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import classNames from 'classnames';
+import {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { getTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Filter } from './components/Filter';
+import { TodoList } from './components/TodoList';
+
+import { FilterStatus } from './types/Filter';
+import { Todo } from './types/Todo';
+
+const getTodoById = (todos: Todo[], todoId: number) => {
+  return todos.find(({ id }) => id === todoId) || null;
+};
+
+const filterTodos = (todos: Todo[], filterStatus: FilterStatus) => {
+  switch (filterStatus) {
+    case 'all': return todos;
+    case 'active': return todos.filter(({ completed }) => !completed);
+    case 'completed': return todos.filter(({ completed }) => completed);
+    default: throw new Error('Error: Filter todos');
+  }
+};
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterStatus, setFilterStatus] = useState<FilterStatus>('all');
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>(todos);
+  const [haveError, setHaveError] = useState(false);
+
+  const handleStatusChange = (todoId: number) => {
+    const copyTodos = [...visibleTodos];
+    const changedTodo = getTodoById(copyTodos, todoId);
+
+    if (changedTodo) {
+      changedTodo.completed = !changedTodo.completed;
+      setVisibleTodos(copyTodos);
+    }
+  };
```

Comment:
Todo changes should be in the next task. 
I also added and changed todo manually without sending it to the server to check the task. But this method would be necessary to remove for review.